bp_be/src/include/bp_be_dcache/bp_be_dcache_pipeline.vh

```diff
@@ -0,0 +1,69 @@
+/**
+ *  Name: 
+ *    bp_be_dcache_pipeline.vh
+ *
+ *  Description: 
+ *    Structs used for dcache pipeline
+ */
+
+`ifndef BP_BE_DCACHE_PIPELINE_VH
+`define BP_BE_DCACHE_PIPELINE_VH
+
+`define declare_bp_amo_fetch_arithmetic_op_s                                            \
+  typedef struct packed {                                                          \
+    logic add_op;                                                               \
+    logic min_op;                                                               \
+    logic max_op;                                                               \
+    logic minu_op;                                                              \
+    logic maxu_op;                                                              \
+  } bp_amo_fetch_arithmetic_op_s;
+
+`define bp_amo_fetch_arithmetic_op_width \
+  (1 + 1 + 1 + 1 + 1)
+
+`define declare_bp_amo_fetch_logic_op_s                                       \
+  typedef struct packed {                                                          \
+    logic xor_op;                                                               \
+    logic and_op;                                                               \
+    logic or_op;                                                                \
+  } bp_amo_fetch_logic_op_s;
+
+`define bp_amo_fetch_logic_op_width \
+  (1 + 1 + 1)
+
+`define declare_bp_be_dcache_size_op_s                                             \
+  typedef struct packed {                                                          \
+    logic double_op;                                                               \
+    logic word_op;                                                                 \
+    logic half_op;                                                                 \
+    logic byte_op;                                                                 \
+  } bp_be_dcache_size_op_s;
+
+`define bp_be_dcache_size_op_width \
+  (1 + 1 + 1 + 1)
+
+`define declare_bp_be_dcache_pipeline_struct_s                                     \
+  typedef struct packed {                                                          \
+    logic load_op;                                                                 \
+    logic store_op;                                                                \
+    logic signed_op;                                                               \
+    logic lr_op;                                                                   \
+    logic sc_op;                                                                   \
+    logic swap_op;                                                              \
+    bp_amo_fetch_logic_op_s amo_fetch_logic;                                       \
```

Comment:
I don't think we gain much from the nested structs, I'd say just flatten the whole thing

Reply:
Sure. However, the size struct may be useful since it packs all four signals as a one hot encoding?

Reply:
The thing is that the struct doesn't actually enforce that one-hot condition